apps/medon-fe/src/components/ProfileForm/index.tsx

```diff
@@ -0,0 +1,284 @@
+            <InputContainer>
+              <Label htmlFor="city">
+                <LabelText>{t('profileForm.city.label')}</LabelText>
+                <Controller
+                  name="city"
+                  control={control}
+                  render={({ field }) => (
+                    <Input
+                      id="city"
+                      size="large"
+                      style={disabled ? AntInputDisabledStyle : AntInputStyle}
+                      disabled={disabled}
+                      status={errors.city?.message ? 'error' : undefined}
+                      placeholder={`${t('profileForm.city.placeholder')}`}
+                      {...field}
+                    />
+                  )}
+                />
+                {errors.city?.message && (
+                  <ErrorMsg role="alert">
+                    {t(`${errors.city?.message}`)}
+                  </ErrorMsg>
+                )}
+              </Label>
+            </InputContainer>
+            <InputContainer>
+              <Label htmlFor="timezone">
+                <ProfileSelect
+                  name={TIMEZONE}
+                  control={control}
+                  disabled={disabled}
+                  error={errors.timezone?.message}
+                  options={timezoneOptions}
+                />
+              </Label>
+            </InputContainer>
+            <InputContainer>
+              <Label htmlFor="role">
+                <ProfileSelect
+                  name={ROLE}
+                  control={control}
+                  disabled={disabled}
+                  error={errors.role?.message}
+                  options={ROLE_OPTIONS}
+                />
+              </Label>
+            </InputContainer>
+            {role === ROLES.REMOTE && (
+              <InputContainer>
+                <Label htmlFor="role">
+                  <ProfileSelect
+                    name={SPECIALITY}
+                    control={control}
+                    error={errors.speciality?.message}
+                    options={SPECIALITY_OPTIONS}
+                    disabled={disabled}
+                  />
+                </Label>
+              </InputContainer>
+            )}
+          </Form>
+          <ButtonContainer>
+            {disabled ? (
+              <StyledButton
+                size="large"
+                htmlType="button"
+                onClick={() => {
+                  setDisabled(false);
+                }}
+              >
+                Edit Profile
+              </StyledButton>
+            ) : (
+              <StyledButton size="large" htmlType="submit">
+                {t('profileForm.profileBtn')}
+              </StyledButton>
+            )}
+            <StyledButton size="large" htmlType="submit" disabled={true}>
+              {t('profileForm.changePasswordBtn')}
+            </StyledButton>
+          </ButtonContainer>
+        </>
+      )}
+    </Container>
+  );
+}
```

Comment:
component is to big, split, logic move to hooks, create reusable component where possible or just smaller one

Reply:
moved all inputs in separate components, no logic in component